### PR TITLE
feat: added catalogId and displayPath support

### PR DESCRIPTION
**Pull request by CL-SHLOMIKONCHA**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✖                                                                        |
| New feature     | ✔                                                                        |
| Breaking change | ✔                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✖                                                                        |
| Tests added     | ✖                                                                        |
| Chore            | ✖                                                                       |

Add tiles path field to catalog record

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -44,7 +44,7 @@
   "dependencies": {
     "@map-colonies/error-express-handler": "^1.2.0",
     "@map-colonies/mc-logger": "^2.0.2",
-    "@map-colonies/mc-model-types": "^12.1.0",
+    "@map-colonies/mc-model-types": "file:../mc-models/map-colonies-mc-model-types-13.1.3.tgz",
```
Comment (asafMasa) [inline package.json:47]: update to real package before merging

File: src/layers/models/layersManager.ts [hunk 2]
```diff
@@ -198,4 +205,20 @@ export class LayersManager {
       });
     }
   }
+
+  private async generateRecordIds(): Promise<IRecordIds> {
+    let id: string;
```
Comment (asafMasa) [inline src/layers/models/layersManager.ts:210]: 1. Add generate catalog id log
2. Add try catch and on exception throw failed to generate catalog id message

File: src/layers/models/layersManager.ts [hunk 3]
```diff
@@ -198,4 +205,20 @@ export class LayersManager {
...
+    let isExists: boolean;
+    do {
+      id = uuidv4();
+      isExists = await this.catalog.existsByRecordId(id);
```
Comment (asafMasa) [inline src/layers/models/layersManager.ts:214]: probably overkill, but add validation against in progress / pending "new" ingestion jobs "catalogId fields"

**Review by asafMasa — CHANGES_REQUESTED** [review]

File: src/layers/models/layersManager.ts (hunk 3 above)
Comment (asafMasa) [reply src/layers/models/layersManager.ts:214]: Fix lint and tests

File: src/layers/models/layersManager.ts [hunk 4]
```diff
@@ -198,4 +206,31 @@ export class LayersManager {
...
     }
   }
+
+  private async generateRecordIds(): Promise<IRecordIds> {
+    let id: string;
+    let isExists: boolean;
+    let jobs: IGetJobResponse[];
+    try {
+      do {
+        this.logger.log('info', `generating record id`);
```
Comment (cl-shaharshavit-rnd) [inline src/layers/models/layersManager.ts:216]: should probably be debug log

File: package.json (hunk 1 above)
Comment (cl-shaharshavit-rnd) [inline package.json:47]: don't forget to replace that with published version

**Review by cl-shaharshavit-rnd — CHANGES_REQUESTED** [review]

File: src/serviceClients/sqliteClient.ts [hunk 5]
```diff
@@ -1,26 +1,27 @@
...
   private readonly fullPath: string;
   private readonly layerSourcesPath: string;
-  private readonly logger: ILogger;
-  private readonly config: IConfig;
 
-  public constructor(packageName: string, originDirectory: string) {
-    this.logger = container.resolve(Services.LOGGER);
-    this.config = container.resolve(Services.CONFIG);
+  public constructor(
+    @inject(Services.CONFIG) private readonly config: IConfig,
```
Comment (cl-shaharshavit-rnd) [inline src/serviceClients/sqliteClient.ts:20]: remove injects

File: src/serviceClients/sqliteClient.ts [hunk 6]
```diff
@@ -1,26 +1,27 @@
...
-import { ILogger } from '../common/interfaces';
+import { ILogger, IConfig } from '../common/interfaces';
 import { Grid } from '../layers/interfaces';
 
 interface IMatrixValues {
   matrixWidth: number;
   matrixHeight: number;
 }
 
+@singleton()
```
Comment (cl-shaharshavit-rnd) [inline src/serviceClients/sqliteClient.ts:13]: remove singleton

File: tests/unit/serviceClients/sqliteClient.spec.ts [hunk 7]
```diff
@@ -56,8 +53,8 @@ describe('SQLClient', () => {
         return { get: () => mockMatrixValues } as Statement;
       });
 
-      const testDb = new SQLiteClient('test.gpkg', '/here');
-      const result = testDb.getGrid();
+      //const testDb = new SQLiteClient(config, logger, 'test.gpkg', '/here');
```
Comment (cl-shaharshavit-rnd) [inline tests/unit/serviceClients/sqliteClient.spec.ts:56]: remove

**Review by cl-shaharshavit-rnd — CHANGES_REQUESTED** [review]

**Review by cl-shaharshavit-rnd — APPROVED** [review]

**Review by asafMasa — APPROVED** [review]

**Pull request merged**